tools/java/org/unicode/cldr/util/SubtypeToURLMap.java

```diff
@@ -0,0 +1,138 @@
+package org.unicode.cldr.util;
+
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.Charset;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+
+import org.unicode.cldr.test.CheckCLDR.CheckStatus.Subtype;
+
+@CLDRTool(alias="subtype-to-url-map", description = "parse each of the params as a subtype map and check, otherwise print the default map")
+public class SubtypeToURLMap {
+
+    /**
+     * Path of the default mapping file
+     */
+    public static final String DEFAULT_MAPPING = "subtypeMapping.txt";
+    private Map<Subtype, String> map;
+
+
+    public SubtypeToURLMap(BufferedReader utf8Data) throws IllegalArgumentException{
+        final Map<Subtype, String> newMap = new HashMap<>();
+        int n = 0;
+        try {
+            int urlLast = 0;
+            int subtypeLast = 0;
+            String url = null;
+            int urlCount = 0;
+            for(String ln; (ln=utf8Data.readLine())!=null;) {
```

Comment:
How about a subroutine for the "for" loop?

Reply:
Not sure how that would help here - this function consists of several state variables, and then the loop, and then several lines of error checking which all depend on the state variables.